bin/data_frame_analyzer/Main.cc

```diff
@@ -145,6 +140,17 @@ int main(int argc, char** argv) {
 
     auto analysisSpecification =
         std::make_unique<ml::api::CDataFrameAnalysisSpecification>(analysisSpecificationJson);
+
+    if (memoryUsageEstimationOnly) {
+        const auto result = analysisSpecification->estimateMemoryUsage();
+        auto outStream = [&ioMgr]() {
+            return std::make_unique<ml::core::CJsonOutputStreamWrapper>(ioMgr.outputStream());
+        }();
+        ml::api::CMemoryUsageEstimationResultJsonWriter jsonWriter(*outStream);
+        jsonWriter.write(result);
```

Comment:
I think it might be nicer to move logic in main for writing the result, i.e.
```
auto outStream = [&ioMgr]() {
            return std::make_unique<ml::core::CJsonOutputStreamWrapper>(ioMgr.outputStream());
        }();
        ml::api::CMemoryUsageEstimationResultJsonWriter jsonWriter(*outStream);
        jsonWriter.write(result);
```
into the function on `CDataFrameAnalysisRunner` then you avoid the need for `SMemoryUsageEstimationResult` altogether. This is also more consistent with the rest of the code which passes off the actual result writing the objects.